worker/worker-execution/score-worker-execution-impl/src/main/java/io/cloudslang/worker/execution/services/ExecutionServiceImpl.java

```diff
@@ -475,42 +476,38 @@ private static boolean isDebuggerMode(Map<String, Serializable> systemContext) {
     }
 
     private void dumpBusEvents(Execution execution) throws InterruptedException {
-        ArrayDeque<ScoreEvent> eventsQueue = execution.getSystemContext().getEvents();
-        if (eventsQueue == null) {
-            return;
-        }
-        for (ScoreEvent eventWrapper : eventsQueue) {
-            eventBus.dispatch(eventWrapper);
+        final ArrayDeque<ScoreEvent> eventsQueue = execution.getSystemContext().getEvents();
+        if ((eventsQueue != null) && !eventsQueue.isEmpty()) {
+            for (ScoreEvent eventWrapper : eventsQueue) {
+                eventBus.dispatch(eventWrapper);
+            }
+            eventsQueue.clear();
         }
-        eventsQueue.clear();
     }
 
     protected ExecutionStep loadExecutionStep(Execution execution) {
         RunningExecutionPlan runningExecutionPlan;
-        if (execution != null) {
-            // Optimization for external workers - run the content only without loading the execution plan
-            if (execution.getSystemContext().get(TempConstants.CONTENT_EXECUTION_STEP) != null) {
-                return (ExecutionStep) execution.getSystemContext().get(TempConstants.CONTENT_EXECUTION_STEP);
-            }
+        // Optimization for external workers - run the content only without loading the execution plan
+        if (execution.getSystemContext().get(TempConstants.CONTENT_EXECUTION_STEP) != null) {
+            return (ExecutionStep) execution.getSystemContext().get(TempConstants.CONTENT_EXECUTION_STEP);
+        } else {
             Long position = execution.getPosition();
             if (position != null) {
-                runningExecutionPlan = workerDbSupportService
-                        .readExecutionPlanById(execution.getRunningExecutionPlanId());
+                runningExecutionPlan = workerDbSupportService.readExecutionPlanById(execution.getRunningExecutionPlanId());
                 if (runningExecutionPlan != null) {
                     updateMetadata(execution, runningExecutionPlan);
                     ExecutionStep currStep = runningExecutionPlan.getExecutionPlan().getStep(position);
                     if (logger.isDebugEnabled()) {
-                        logger.debug("Begin step: " + position + " in flow " + runningExecutionPlan.getExecutionPlan()
-                                .getFlowUuid() + " [" + execution.getExecutionId() + "]");
+                        logger.debug("Begin step: " + position + " in flow " + runningExecutionPlan.getExecutionPlan().getFlowUuid() + " [" + execution.getExecutionId() + "]");
```

Comment:
Even though the previous "aligned" line still crosses "the line", this one is over it by A LOT. It would be better to leave it as it was.
Or replace it with String.format :)